src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+section.
+
+1. Enable the [IAM Service Account Credentials API][] in the cloud
+   project. Attempts to impersonate the service account fail without
+   this API.
+
+   ```console
+   gcloud services enable iamcredentials.googleapis.com
+   ```
+
+1. Find the project number.
+
+   ```console
+   gcloud projects describe $PROJECT_ID --format='value(projectNumber)'
+   ```
+
+1. Grant permission to impersonate the publishing service account.
+
+   ```console
+   gcloud iam service-accounts add-iam-policy-binding \
+     pub-dev@$PROJECT_ID.iam.gserviceaccount.com \
+     --member=serviceAccount:$PROJECT_NUMBER@cloudbuild.gserviceaccount.com \
+     --role=roles/iam.serviceAccountTokenCreator
+   ```
+
+### Create a Cloud Build Configuration File
+
+To publish from Cloud Build, enable Cloud Build to:
+
+* Impersonate the service account to obtain a temporary OIDC `id_token`
+* Provide the temporary `id_token` to `dart pub` for use when publishing
+* Call `dart pub publish` to publish the package
+
+You specify these steps for Google Cloud Build in a `cloudbuild.yaml`
+file. To learn more about the format of this file, check out
+[build configuration file schema][15].
+
+A Cloud Build configuration file that works with this example should
+resemble the following YAML markup:
+
+```yaml
+# cloudbuild.yaml
+steps:
+- id: Create temporary token
+  name: gcr.io/cloud-builders/gcloud
+  volumes:
+  - name: temporary-secrets
+    path: /secrets
+  script: |
+    gcloud auth print-identity-token \
+      --impersonate-service-account=pub-dev@$PROJECT_ID.iam.gserviceaccount.com \
+      --audiences=https://pub.dev \
+      --include-email > /secrets/temporary-pub-token.txt
+  env:
+  - PROJECT_ID=$PROJECT_ID
+- id: Publish to pub.dev
+  name: dart
+  volumes:
+  - name: temporary-secrets
+    path: /secrets
+  script: | 
+    cat /secrets/temporary-pub-token.txt | dart pub token add https://pub.dev
+    dart pub publish --force
+```
+
+The `Create temporary token` step features the
+`gcloud auth print-identity-token` script.
+This script performs the following actions:
+
+* It creates an OpenID Connect `id_token`.
+  Google signs this `id_token`. The signature that expires within one
+  hour.
+* The `--impersonate-service-account` option identifies the service
+  account to impersonate. This example uses the service account created
+  in a previous activity in this guide.
+* The `--audiences` option identifies `pub.dev` as the intended
+  recipient of the token. 
+* The `--include-email` option requests that `pub.dev` recognize the
+  service account.
+
+Once you create the `id_token`, Google Cloud writes it to a file that
+resides in a _volume_. Google Cloud uses this mechanism to
+[pass data between steps][16].
+Don't store the token in `/workspace`.
+Google Cloud checks out your repository into `/workspace`.
+If you store the token in `/workspace`, your published package includes
+the `id_token`.
```

Comment:
That's a strong an unsubstantiated statement.

It may, if you don't delete it or remove it, but it's not a certainty.
Or rather, I'm unwilling to formally prove that this is always the case.